src/tox/_pytestplugin.py

```diff
@@ -176,9 +176,13 @@ def outlines(self):
         return err + out
 
     def __repr__(self):
-        return "RunResult(ret={}, args={}, out=\n{}\n, err=\n{})".format(
-            self.ret, " ".join(str(i) for i in self.args), self.out, self.err
+        res = "RunResult(ret={}, args={!r}, out=\n{}\n, err=\n{})".format(
+            self.ret, self.args, self.out, self.err
         )
+        if six.PY2:
+            return res.encode("UTF-8")
+        else:
+            return res
```

Comment:
can you make it one return?

Reply:
one less line 

```
        if six.PY2:
            res = res.encode("UTF-8")
        return res
```

Reply:
`if` `else` is easier for humans to parse as it's clearer that there are two disparate branches -- for maintenance reasons it's also easier to update if functionality were to be added as it's clearer it needs to happen at two places at a glance.

I've seen too many bugs from using your suggested style 😆

Reply:
something must be wrong with me then because for me two returns are a lot harder to understand, reason and parse 🤔 I think it's harder to maintain as now I need to update in two locations while I can clearly reduce the second to one, needing to worry only about a single case...

Reply:
your suggested pattern also makes mypy unhappy since it changes the type of the variable

Reply:
works in my projects with mypy 🤔

Reply:
Seems our test is broken, can you fix it? I think on Windows py27 the order is inverted or undefined 🤔

Reply:
yeah happy to hack on the flakey test -- I knew there was one just the cascade of pytest / repr / etc. bugs was hiding it :laughing:

Reply:
fixed it on my PR 👍 now only to figure out why coverage html is not getting published

Reply:
lol it's always something >.<

Reply:
it worked a while back now it's missing 💭

Reply:
https://github.com/Microsoft/azure-pipelines-tasks/pull/9962 turns out the ability to upload coverage report was removed 😆

Reply:
now is auto-generated instead https://docs.microsoft.com/en-us/azure/devops/release-notes/2019/sprint-150-update#cobertura-code-coverage-report-updates

Reply:
whoa that's cool